.github/workflows/Tests.yaml

```diff
@@ -394,3 +394,174 @@ jobs:
+          echo "    buildable: False" >> $SPACK_PACKAGES
+          cat $SPACK_PACKAGES
+          git clone https://github.com/spack/spack.git
+          source ./spack/share/spack/setup-env.sh
+          spack compiler find --scope defaults
+          spack mirror add dependencies file://$HOME/dependencies/spack
+          spack env create spectre
+          spack env activate spectre
+          spack install -j4 --no-check-signature $SPECTRE_SPACK_DEPS
+          spack find -v
+      # After building the dependencies from source we cache them as compressed
+      # tarballs.
+      - name: Package built Spack dependencies for caching
+        run: |
+          cd $HOME
+          source ./spack/share/spack/setup-env.sh
+          spack env activate spectre
+          spack buildcache create -uf -m dependencies $SPECTRE_SPACK_DEPS
+      # Install Charm++ separately. We may be able to install it via Spack as
+      # well and patch it.
+      - name: Install Charm++
+        if: steps.restore-dependencies.outputs.cache-hit != 'true'
+        run: |
+          cd $HOME/dependencies
+          wget http://charm.cs.illinois.edu/distrib/charm-${CHARM_VERSION}.tar.bz2
+          tar -xjf charm-${CHARM_VERSION}.tar.bz2
+          cd ./charm-${CHARM_VERSION}
+          ./build charm++ multicore-darwin-x86_64 -j4 -O0 $SPECTRE_MIN_MACOS
+          git apply $GITHUB_WORKSPACE/support/Charm/${CHARM_PATCH}
+          rm ../charm-${CHARM_VERSION}.tar.bz2
+      # Replace the ccache directory that building the dependencies may have
+      # generated with the cached ccache directory.
+      - name: Clear ccache from dependencies
+        run: |
+          rm -rf $CCACHE_DIR
+          mkdir -p $CCACHE_DIR
+      - name: Restore ccache
+        uses: actions/cache@v1
+        with:
+          path: ~/ccache
+          key: ccache-macos
+      - name: Configure ccache
+        run: |
+          ccache -pz
+      # Configure, build and run tests. See the `unit_tests` job for details.
+      # We increase the timeout for tests because `ctest` on the GitHub-hosted
+      # macOS VMs appears to be quite slow.
+      - name: Configure build with cmake
+        run: >
+          mkdir build && cd build
+
+          source $HOME/spack/share/spack/setup-env.sh
+
+          spack env activate spectre
+
+          cmake
+          -D CMAKE_C_COMPILER=clang
+          -D CMAKE_CXX_COMPILER=clang++
+          -D CMAKE_Fortran_COMPILER=gfortran
+          -D CMAKE_CXX_FLAGS="-Werror"
+          -D OVERRIDE_ARCH=x86-64
+          -D CHARM_ROOT=$HOME/dependencies/charm-${CHARM_VERSION}
+          -D BLAS_ROOT=$(brew --prefix openblas)
+          -D MACOSX_MIN=10.11
+          -D CMAKE_BUILD_TYPE=Debug
+          -D DEBUG_SYMBOLS=OFF
+          -D RUN_TESTS_IN_TEST_EXECUTABLES=OFF
+          -D STUB_EXECUTABLE_OBJECT_FILES=ON
+          -D STUB_LIBRARY_OBJECT_FILES=ON
+          -D USE_PCH=ON
+          -D USE_CCACHE=ON
```

Comment:
Given the limited cache space, are we better off not using ccache for spectre in order to favor caching the dependencies? Or is that why the max cache size is 400M?

Reply:
looks like the max total cache size has increased to 5gb, and I think the size per individual cache is also not limited to 400mb anymore. I haven't really understood the eviction policy yet though. In my testing the dependency cache was always available, but that can have a lot of reasons. I opened https://github.com/sxs-collaboration/spectre/issues/2173 to look into this.

Reply:
Sounds good, thanks!